pkg/cortex/modules.go

```diff
@@ -309,6 +310,10 @@ func (t *Cortex) initIngester() (serv services.Service, err error) {
 		return
 	}
 
+	return t.Ingester, nil
+}
+
+func (t *Cortex) initIngester() (serv services.Service, err error) {
 	t.API.RegisterIngester(t.Ingester, t.Cfg.Distributor)
 
 	return t.Ingester, nil
```

Comment:
This needs to return `nil` as a service, otherwise `t.Ingester` will be tried to be started twice (once from `initIngester`, once from `initIngesterService`).

```suggestion
	return nil, nil
```

(I tried to pushing to your branch, but that failed. Probably for the better :))

Reply:
ah, thx!